[2024-11-09T23:13:35.081+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-11-09T23:13:35.097+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: openbrewerydb_dag.spark_submit_bronze_job manual__2024-11-09T22:55:57.289669+00:00 [queued]>
[2024-11-09T23:13:35.104+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: openbrewerydb_dag.spark_submit_bronze_job manual__2024-11-09T22:55:57.289669+00:00 [queued]>
[2024-11-09T23:13:35.104+0000] {taskinstance.py:2865} INFO - Starting attempt 5 of 5
[2024-11-09T23:13:35.118+0000] {taskinstance.py:2888} INFO - Executing <Task(SparkSubmitOperator): spark_submit_bronze_job> on 2024-11-09 22:55:57.289669+00:00
[2024-11-09T23:13:35.129+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=63) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-11-09T23:13:35.130+0000] {standard_task_runner.py:72} INFO - Started process 64 to run task
[2024-11-09T23:13:35.130+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'openbrewerydb_dag', 'spark_submit_bronze_job', 'manual__2024-11-09T22:55:57.289669+00:00', '--job-id', '482', '--raw', '--subdir', 'DAGS_FOLDER/openbrewerydb_dag.py', '--cfg-path', '/tmp/tmpllfg2ojw']
[2024-11-09T23:13:35.131+0000] {standard_task_runner.py:105} INFO - Job 482: Subtask spark_submit_bronze_job
[2024-11-09T23:13:35.182+0000] {task_command.py:467} INFO - Running <TaskInstance: openbrewerydb_dag.spark_submit_bronze_job manual__2024-11-09T22:55:57.289669+00:00 [running]> on host 3ba6f6ff617e
[2024-11-09T23:13:35.279+0000] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='openbrewerydb_dag' AIRFLOW_CTX_TASK_ID='spark_submit_bronze_job' AIRFLOW_CTX_EXECUTION_DATE='2024-11-09T22:55:57.289669+00:00' AIRFLOW_CTX_TRY_NUMBER='5' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-11-09T22:55:57.289669+00:00'
[2024-11-09T23:13:35.280+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-11-09T23:13:35.322+0000] {base.py:84} INFO - Retrieving connection 'spark_default'
[2024-11-09T23:13:35.326+0000] {spark_submit.py:473} INFO - Spark-Submit cmd: spark-submit --master spark-master:7077 --jars /opt/***/dags/scripts/jars/hadoop-azure-3.4.1.jar,/opt/***/dags/scripts/jars/azure-storage-8.6.6.jar,/opt/***/dags/scripts/jars/jetty-server-9.4.45.v20220203.jar,/opt/***/dags/scripts/jars/jetty-util-9.4.45.v20220203.jar,/opt/***/dags/scripts/jars/jetty-util-ajax-9.4.45.v20220203.jar --name spark_submit_bronze_task --deploy-mode client /opt/***/dags/scripts/bronze_layer.py
[2024-11-09T23:13:37.006+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:37 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2024-11-09T23:13:39.182+0000] {spark_submit.py:634} INFO - 2024-11-09 23:13:39,179 - INFO - Pipeline execution started.
[2024-11-09T23:13:39.183+0000] {spark_submit.py:634} INFO - 2024-11-09 23:13:39,179 - INFO - Initializing Spark session.
[2024-11-09T23:13:39.334+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:39 INFO SparkContext: Running Spark version 3.5.3
[2024-11-09T23:13:39.334+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:39 INFO SparkContext: OS info Linux, 6.10.13-3-MANJARO, amd64
[2024-11-09T23:13:39.335+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:39 INFO SparkContext: Java version 17.0.13
[2024-11-09T23:13:39.445+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:39 INFO ResourceUtils: ==============================================================
[2024-11-09T23:13:39.446+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:39 INFO ResourceUtils: No custom resources configured for spark.driver.
[2024-11-09T23:13:39.446+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:39 INFO ResourceUtils: ==============================================================
[2024-11-09T23:13:39.446+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:39 INFO SparkContext: Submitted application: Brewery Data Pipeline
[2024-11-09T23:13:39.495+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:39 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(cores -> name: cores, amount: 1, script: , vendor: , memory -> name: memory, amount: 1024, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2024-11-09T23:13:39.510+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:39 INFO ResourceProfile: Limiting resource is cpu
[2024-11-09T23:13:39.511+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:39 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2024-11-09T23:13:39.604+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:39 INFO SecurityManager: Changing view acls to: ***
[2024-11-09T23:13:39.608+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:39 INFO SecurityManager: Changing modify acls to: ***
[2024-11-09T23:13:39.608+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:39 INFO SecurityManager: Changing view acls groups to:
[2024-11-09T23:13:39.609+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:39 INFO SecurityManager: Changing modify acls groups to:
[2024-11-09T23:13:39.609+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:39 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users with view permissions: ***; groups with view permissions: EMPTY; users with modify permissions: ***; groups with modify permissions: EMPTY
[2024-11-09T23:13:40.083+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:40 INFO Utils: Successfully started service 'sparkDriver' on port 34845.
[2024-11-09T23:13:40.130+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:40 INFO SparkEnv: Registering MapOutputTracker
[2024-11-09T23:13:40.189+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:40 INFO SparkEnv: Registering BlockManagerMaster
[2024-11-09T23:13:40.220+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:40 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2024-11-09T23:13:40.222+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:40 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2024-11-09T23:13:40.234+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:40 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2024-11-09T23:13:40.273+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:40 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-8d749a0a-a9b4-45f1-a656-913ca02d5f4b
[2024-11-09T23:13:40.312+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:40 INFO MemoryStore: MemoryStore started with capacity 434.4 MiB
[2024-11-09T23:13:40.356+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:40 INFO SparkEnv: Registering OutputCommitCoordinator
[2024-11-09T23:13:40.643+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:40 INFO JettyUtils: Start Jetty 0.0.0.0:4040 for SparkUI
[2024-11-09T23:13:40.774+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:40 INFO Utils: Successfully started service 'SparkUI' on port 4040.
[2024-11-09T23:13:40.836+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:40 INFO SparkContext: Added JAR file:///opt/***/dags/scripts/jars/hadoop-azure-3.4.1.jar at spark://3ba6f6ff617e:34845/jars/hadoop-azure-3.4.1.jar with timestamp 1731194019310
[2024-11-09T23:13:40.841+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:40 INFO SparkContext: Added JAR file:///opt/***/dags/scripts/jars/azure-storage-8.6.6.jar at spark://3ba6f6ff617e:34845/jars/azure-storage-8.6.6.jar with timestamp 1731194019310
[2024-11-09T23:13:40.842+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:40 INFO SparkContext: Added JAR file:///opt/***/dags/scripts/jars/jetty-server-9.4.45.v20220203.jar at spark://3ba6f6ff617e:34845/jars/jetty-server-9.4.45.v20220203.jar with timestamp 1731194019310
[2024-11-09T23:13:40.843+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:40 INFO SparkContext: Added JAR file:///opt/***/dags/scripts/jars/jetty-util-9.4.45.v20220203.jar at spark://3ba6f6ff617e:34845/jars/jetty-util-9.4.45.v20220203.jar with timestamp 1731194019310
[2024-11-09T23:13:40.846+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:40 INFO SparkContext: Added JAR file:///opt/***/dags/scripts/jars/jetty-util-ajax-9.4.45.v20220203.jar at spark://3ba6f6ff617e:34845/jars/jetty-util-ajax-9.4.45.v20220203.jar with timestamp 1731194019310
[2024-11-09T23:13:40.936+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:40 INFO Executor: Starting executor ID driver on host 3ba6f6ff617e
[2024-11-09T23:13:40.937+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:40 INFO Executor: OS info Linux, 6.10.13-3-MANJARO, amd64
[2024-11-09T23:13:40.937+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:40 INFO Executor: Java version 17.0.13
[2024-11-09T23:13:40.946+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:40 INFO Executor: Starting executor with user classpath (userClassPathFirst = false): ''
[2024-11-09T23:13:40.947+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:40 INFO Executor: Created or updated repl class loader org.apache.spark.util.MutableURLClassLoader@4b2af96d for default.
[2024-11-09T23:13:40.961+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:40 INFO Executor: Fetching spark://3ba6f6ff617e:34845/jars/jetty-server-9.4.45.v20220203.jar with timestamp 1731194019310
[2024-11-09T23:13:41.019+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:41 INFO TransportClientFactory: Successfully created connection to 3ba6f6ff617e/172.23.0.9:34845 after 30 ms (0 ms spent in bootstraps)
[2024-11-09T23:13:41.031+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:41 INFO Utils: Fetching spark://3ba6f6ff617e:34845/jars/jetty-server-9.4.45.v20220203.jar to /tmp/spark-2159c945-ced0-4794-a5b6-7b6a63fb0c2b/userFiles-baf26796-5703-457d-9abd-045ef993e1a0/fetchFileTemp9979826732678359823.tmp
[2024-11-09T23:13:41.079+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:41 INFO Executor: Adding file:/tmp/spark-2159c945-ced0-4794-a5b6-7b6a63fb0c2b/userFiles-baf26796-5703-457d-9abd-045ef993e1a0/jetty-server-9.4.45.v20220203.jar to class loader default
[2024-11-09T23:13:41.080+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:41 INFO Executor: Fetching spark://3ba6f6ff617e:34845/jars/azure-storage-8.6.6.jar with timestamp 1731194019310
[2024-11-09T23:13:41.081+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:41 INFO Utils: Fetching spark://3ba6f6ff617e:34845/jars/azure-storage-8.6.6.jar to /tmp/spark-2159c945-ced0-4794-a5b6-7b6a63fb0c2b/userFiles-baf26796-5703-457d-9abd-045ef993e1a0/fetchFileTemp16738530180632790387.tmp
[2024-11-09T23:13:41.121+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:41 INFO Executor: Adding file:/tmp/spark-2159c945-ced0-4794-a5b6-7b6a63fb0c2b/userFiles-baf26796-5703-457d-9abd-045ef993e1a0/azure-storage-8.6.6.jar to class loader default
[2024-11-09T23:13:41.123+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:41 INFO Executor: Fetching spark://3ba6f6ff617e:34845/jars/jetty-util-9.4.45.v20220203.jar with timestamp 1731194019310
[2024-11-09T23:13:41.123+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:41 INFO Utils: Fetching spark://3ba6f6ff617e:34845/jars/jetty-util-9.4.45.v20220203.jar to /tmp/spark-2159c945-ced0-4794-a5b6-7b6a63fb0c2b/userFiles-baf26796-5703-457d-9abd-045ef993e1a0/fetchFileTemp13289357795043606111.tmp
[2024-11-09T23:13:41.139+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:41 INFO Executor: Adding file:/tmp/spark-2159c945-ced0-4794-a5b6-7b6a63fb0c2b/userFiles-baf26796-5703-457d-9abd-045ef993e1a0/jetty-util-9.4.45.v20220203.jar to class loader default
[2024-11-09T23:13:41.140+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:41 INFO Executor: Fetching spark://3ba6f6ff617e:34845/jars/jetty-util-ajax-9.4.45.v20220203.jar with timestamp 1731194019310
[2024-11-09T23:13:41.154+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:41 INFO Utils: Fetching spark://3ba6f6ff617e:34845/jars/jetty-util-ajax-9.4.45.v20220203.jar to /tmp/spark-2159c945-ced0-4794-a5b6-7b6a63fb0c2b/userFiles-baf26796-5703-457d-9abd-045ef993e1a0/fetchFileTemp17410807789652925885.tmp
[2024-11-09T23:13:41.169+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:41 INFO Executor: Adding file:/tmp/spark-2159c945-ced0-4794-a5b6-7b6a63fb0c2b/userFiles-baf26796-5703-457d-9abd-045ef993e1a0/jetty-util-ajax-9.4.45.v20220203.jar to class loader default
[2024-11-09T23:13:41.172+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:41 INFO Executor: Fetching spark://3ba6f6ff617e:34845/jars/hadoop-azure-3.4.1.jar with timestamp 1731194019310
[2024-11-09T23:13:41.173+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:41 INFO Utils: Fetching spark://3ba6f6ff617e:34845/jars/hadoop-azure-3.4.1.jar to /tmp/spark-2159c945-ced0-4794-a5b6-7b6a63fb0c2b/userFiles-baf26796-5703-457d-9abd-045ef993e1a0/fetchFileTemp9923855080193851786.tmp
[2024-11-09T23:13:41.201+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:41 INFO Executor: Adding file:/tmp/spark-2159c945-ced0-4794-a5b6-7b6a63fb0c2b/userFiles-baf26796-5703-457d-9abd-045ef993e1a0/hadoop-azure-3.4.1.jar to class loader default
[2024-11-09T23:13:41.252+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:41 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 44823.
[2024-11-09T23:13:41.253+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:41 INFO NettyBlockTransferService: Server created on 3ba6f6ff617e:44823
[2024-11-09T23:13:41.257+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:41 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2024-11-09T23:13:41.286+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:41 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 3ba6f6ff617e, 44823, None)
[2024-11-09T23:13:41.302+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:41 INFO BlockManagerMasterEndpoint: Registering block manager 3ba6f6ff617e:44823 with 434.4 MiB RAM, BlockManagerId(driver, 3ba6f6ff617e, 44823, None)
[2024-11-09T23:13:41.304+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:41 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 3ba6f6ff617e, 44823, None)
[2024-11-09T23:13:41.308+0000] {spark_submit.py:634} INFO - 24/11/09 23:13:41 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 3ba6f6ff617e, 44823, None)
[2024-11-09T23:13:41.925+0000] {spark_submit.py:634} INFO - 2024-11-09 23:13:41,923 - INFO - Fetching brewery data from the API.
[2024-11-09T23:15:11.175+0000] {spark_submit.py:634} INFO - Traceback (most recent call last):
[2024-11-09T23:15:11.175+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/urllib3/connectionpool.py", line 789, in urlopen
[2024-11-09T23:15:11.176+0000] {spark_submit.py:634} INFO - response = self._make_request(
[2024-11-09T23:15:11.176+0000] {spark_submit.py:634} INFO - ^^^^^^^^^^^^^^^^^^^
[2024-11-09T23:15:11.176+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/urllib3/connectionpool.py", line 490, in _make_request
[2024-11-09T23:15:11.176+0000] {spark_submit.py:634} INFO - raise new_e
[2024-11-09T23:15:11.176+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/urllib3/connectionpool.py", line 466, in _make_request
[2024-11-09T23:15:11.176+0000] {spark_submit.py:634} INFO - self._validate_conn(conn)
[2024-11-09T23:15:11.176+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/urllib3/connectionpool.py", line 1095, in _validate_conn
[2024-11-09T23:15:11.177+0000] {spark_submit.py:634} INFO - conn.connect()
[2024-11-09T23:15:11.177+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/urllib3/connection.py", line 652, in connect
[2024-11-09T23:15:11.177+0000] {spark_submit.py:634} INFO - sock_and_verified = _ssl_wrap_socket_and_match_hostname(
[2024-11-09T23:15:11.177+0000] {spark_submit.py:634} INFO - ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2024-11-09T23:15:11.177+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/urllib3/connection.py", line 805, in _ssl_wrap_socket_and_match_hostname
[2024-11-09T23:15:11.177+0000] {spark_submit.py:634} INFO - ssl_sock = ssl_wrap_socket(
[2024-11-09T23:15:11.177+0000] {spark_submit.py:634} INFO - ^^^^^^^^^^^^^^^^
[2024-11-09T23:15:11.177+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/urllib3/util/ssl_.py", line 465, in ssl_wrap_socket
[2024-11-09T23:15:11.178+0000] {spark_submit.py:634} INFO - ssl_sock = _ssl_wrap_socket_impl(sock, context, tls_in_tls, server_hostname)
[2024-11-09T23:15:11.178+0000] {spark_submit.py:634} INFO - ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2024-11-09T23:15:11.178+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/urllib3/util/ssl_.py", line 509, in _ssl_wrap_socket_impl
[2024-11-09T23:15:11.178+0000] {spark_submit.py:634} INFO - return ssl_context.wrap_socket(sock, server_hostname=server_hostname)
[2024-11-09T23:15:11.178+0000] {spark_submit.py:634} INFO - ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2024-11-09T23:15:11.178+0000] {spark_submit.py:634} INFO - File "/usr/local/lib/python3.12/ssl.py", line 455, in wrap_socket
[2024-11-09T23:15:11.178+0000] {spark_submit.py:634} INFO - return self.sslsocket_class._create(
[2024-11-09T23:15:11.178+0000] {spark_submit.py:634} INFO - ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2024-11-09T23:15:11.179+0000] {spark_submit.py:634} INFO - File "/usr/local/lib/python3.12/ssl.py", line 1041, in _create
[2024-11-09T23:15:11.179+0000] {spark_submit.py:634} INFO - self.do_handshake()
[2024-11-09T23:15:11.179+0000] {spark_submit.py:634} INFO - File "/usr/local/lib/python3.12/ssl.py", line 1319, in do_handshake
[2024-11-09T23:15:11.179+0000] {spark_submit.py:634} INFO - self._sslobj.do_handshake()
[2024-11-09T23:15:11.179+0000] {spark_submit.py:634} INFO - ConnectionResetError: [Errno 104] Connection reset by peer
[2024-11-09T23:15:11.179+0000] {spark_submit.py:634} INFO - 
[2024-11-09T23:15:11.179+0000] {spark_submit.py:634} INFO - During handling of the above exception, another exception occurred:
[2024-11-09T23:15:11.179+0000] {spark_submit.py:634} INFO - 
[2024-11-09T23:15:11.180+0000] {spark_submit.py:634} INFO - Traceback (most recent call last):
[2024-11-09T23:15:11.180+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/requests/adapters.py", line 667, in send
[2024-11-09T23:15:11.180+0000] {spark_submit.py:634} INFO - resp = conn.urlopen(
[2024-11-09T23:15:11.180+0000] {spark_submit.py:634} INFO - ^^^^^^^^^^^^^
[2024-11-09T23:15:11.180+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/urllib3/connectionpool.py", line 843, in urlopen
[2024-11-09T23:15:11.180+0000] {spark_submit.py:634} INFO - retries = retries.increment(
[2024-11-09T23:15:11.180+0000] {spark_submit.py:634} INFO - ^^^^^^^^^^^^^^^^^^
[2024-11-09T23:15:11.181+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/urllib3/util/retry.py", line 474, in increment
[2024-11-09T23:15:11.181+0000] {spark_submit.py:634} INFO - raise reraise(type(error), error, _stacktrace)
[2024-11-09T23:15:11.181+0000] {spark_submit.py:634} INFO - ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2024-11-09T23:15:11.181+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/urllib3/util/util.py", line 38, in reraise
[2024-11-09T23:15:11.181+0000] {spark_submit.py:634} INFO - raise value.with_traceback(tb)
[2024-11-09T23:15:11.181+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/urllib3/connectionpool.py", line 789, in urlopen
[2024-11-09T23:15:11.181+0000] {spark_submit.py:634} INFO - response = self._make_request(
[2024-11-09T23:15:11.182+0000] {spark_submit.py:634} INFO - ^^^^^^^^^^^^^^^^^^^
[2024-11-09T23:15:11.182+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/urllib3/connectionpool.py", line 490, in _make_request
[2024-11-09T23:15:11.182+0000] {spark_submit.py:634} INFO - raise new_e
[2024-11-09T23:15:11.182+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/urllib3/connectionpool.py", line 466, in _make_request
[2024-11-09T23:15:11.182+0000] {spark_submit.py:634} INFO - self._validate_conn(conn)
[2024-11-09T23:15:11.182+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/urllib3/connectionpool.py", line 1095, in _validate_conn
[2024-11-09T23:15:11.183+0000] {spark_submit.py:634} INFO - conn.connect()
[2024-11-09T23:15:11.183+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/urllib3/connection.py", line 652, in connect
[2024-11-09T23:15:11.183+0000] {spark_submit.py:634} INFO - sock_and_verified = _ssl_wrap_socket_and_match_hostname(
[2024-11-09T23:15:11.183+0000] {spark_submit.py:634} INFO - ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2024-11-09T23:15:11.183+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/urllib3/connection.py", line 805, in _ssl_wrap_socket_and_match_hostname
[2024-11-09T23:15:11.183+0000] {spark_submit.py:634} INFO - ssl_sock = ssl_wrap_socket(
[2024-11-09T23:15:11.183+0000] {spark_submit.py:634} INFO - ^^^^^^^^^^^^^^^^
[2024-11-09T23:15:11.183+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/urllib3/util/ssl_.py", line 465, in ssl_wrap_socket
[2024-11-09T23:15:11.184+0000] {spark_submit.py:634} INFO - ssl_sock = _ssl_wrap_socket_impl(sock, context, tls_in_tls, server_hostname)
[2024-11-09T23:15:11.184+0000] {spark_submit.py:634} INFO - ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2024-11-09T23:15:11.184+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/urllib3/util/ssl_.py", line 509, in _ssl_wrap_socket_impl
[2024-11-09T23:15:11.185+0000] {spark_submit.py:634} INFO - return ssl_context.wrap_socket(sock, server_hostname=server_hostname)
[2024-11-09T23:15:11.185+0000] {spark_submit.py:634} INFO - ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2024-11-09T23:15:11.185+0000] {spark_submit.py:634} INFO - File "/usr/local/lib/python3.12/ssl.py", line 455, in wrap_socket
[2024-11-09T23:15:11.185+0000] {spark_submit.py:634} INFO - return self.sslsocket_class._create(
[2024-11-09T23:15:11.185+0000] {spark_submit.py:634} INFO - ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2024-11-09T23:15:11.186+0000] {spark_submit.py:634} INFO - File "/usr/local/lib/python3.12/ssl.py", line 1041, in _create
[2024-11-09T23:15:11.186+0000] {spark_submit.py:634} INFO - self.do_handshake()
[2024-11-09T23:15:11.186+0000] {spark_submit.py:634} INFO - File "/usr/local/lib/python3.12/ssl.py", line 1319, in do_handshake
[2024-11-09T23:15:11.186+0000] {spark_submit.py:634} INFO - self._sslobj.do_handshake()
[2024-11-09T23:15:11.186+0000] {spark_submit.py:634} INFO - urllib3.exceptions.ProtocolError: ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))
[2024-11-09T23:15:11.187+0000] {spark_submit.py:634} INFO - 
[2024-11-09T23:15:11.187+0000] {spark_submit.py:634} INFO - During handling of the above exception, another exception occurred:
[2024-11-09T23:15:11.187+0000] {spark_submit.py:634} INFO - 
[2024-11-09T23:15:11.187+0000] {spark_submit.py:634} INFO - Traceback (most recent call last):
[2024-11-09T23:15:11.187+0000] {spark_submit.py:634} INFO - File "/opt/***/dags/scripts/bronze_layer.py", line 152, in <module>
[2024-11-09T23:15:11.188+0000] {spark_submit.py:634} INFO - run_pipeline()
[2024-11-09T23:15:11.188+0000] {spark_submit.py:634} INFO - File "/opt/***/dags/scripts/bronze_layer.py", line 123, in run_pipeline
[2024-11-09T23:15:11.188+0000] {spark_submit.py:634} INFO - brewery_data = fetch_brewery_data()
[2024-11-09T23:15:11.188+0000] {spark_submit.py:634} INFO - ^^^^^^^^^^^^^^^^^^^^
[2024-11-09T23:15:11.188+0000] {spark_submit.py:634} INFO - File "/opt/***/dags/scripts/bronze_layer.py", line 60, in fetch_brewery_data
[2024-11-09T23:15:11.189+0000] {spark_submit.py:634} INFO - response = requests.get(API_URL, params={'page': page, 'per_page': 50})
[2024-11-09T23:15:11.189+0000] {spark_submit.py:634} INFO - ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2024-11-09T23:15:11.189+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/requests/api.py", line 73, in get
[2024-11-09T23:15:11.189+0000] {spark_submit.py:634} INFO - return request("get", url, params=params, **kwargs)
[2024-11-09T23:15:11.189+0000] {spark_submit.py:634} INFO - ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2024-11-09T23:15:11.189+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/requests/api.py", line 59, in request
[2024-11-09T23:15:11.189+0000] {spark_submit.py:634} INFO - return session.request(method=method, url=url, **kwargs)
[2024-11-09T23:15:11.190+0000] {spark_submit.py:634} INFO - ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2024-11-09T23:15:11.190+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/requests/sessions.py", line 589, in request
[2024-11-09T23:15:11.190+0000] {spark_submit.py:634} INFO - resp = self.send(prep, **send_kwargs)
[2024-11-09T23:15:11.190+0000] {spark_submit.py:634} INFO - ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2024-11-09T23:15:11.190+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/requests/sessions.py", line 703, in send
[2024-11-09T23:15:11.190+0000] {spark_submit.py:634} INFO - r = adapter.send(request, **kwargs)
[2024-11-09T23:15:11.191+0000] {spark_submit.py:634} INFO - ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2024-11-09T23:15:11.191+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/requests/adapters.py", line 682, in send
[2024-11-09T23:15:11.191+0000] {spark_submit.py:634} INFO - raise ConnectionError(err, request=request)
[2024-11-09T23:15:11.191+0000] {spark_submit.py:634} INFO - requests.exceptions.ConnectionError: ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))
[2024-11-09T23:15:11.192+0000] {spark_submit.py:634} INFO - 2024-11-09 23:15:11,183 - INFO - Closing down clientserver connection
[2024-11-09T23:15:11.419+0000] {spark_submit.py:634} INFO - 24/11/09 23:15:11 INFO SparkContext: Invoking stop() from shutdown hook
[2024-11-09T23:15:11.420+0000] {spark_submit.py:634} INFO - 24/11/09 23:15:11 INFO SparkContext: SparkContext is stopping with exitCode 0.
[2024-11-09T23:15:11.440+0000] {spark_submit.py:634} INFO - 24/11/09 23:15:11 INFO SparkUI: Stopped Spark web UI at http://3ba6f6ff617e:4040
[2024-11-09T23:15:11.451+0000] {spark_submit.py:634} INFO - 24/11/09 23:15:11 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
[2024-11-09T23:15:11.476+0000] {spark_submit.py:634} INFO - 24/11/09 23:15:11 INFO MemoryStore: MemoryStore cleared
[2024-11-09T23:15:11.477+0000] {spark_submit.py:634} INFO - 24/11/09 23:15:11 INFO BlockManager: BlockManager stopped
[2024-11-09T23:15:11.484+0000] {spark_submit.py:634} INFO - 24/11/09 23:15:11 INFO BlockManagerMaster: BlockManagerMaster stopped
[2024-11-09T23:15:11.488+0000] {spark_submit.py:634} INFO - 24/11/09 23:15:11 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
[2024-11-09T23:15:11.499+0000] {spark_submit.py:634} INFO - 24/11/09 23:15:11 INFO SparkContext: Successfully stopped SparkContext
[2024-11-09T23:15:11.500+0000] {spark_submit.py:634} INFO - 24/11/09 23:15:11 INFO ShutdownHookManager: Shutdown hook called
[2024-11-09T23:15:11.501+0000] {spark_submit.py:634} INFO - 24/11/09 23:15:11 INFO ShutdownHookManager: Deleting directory /tmp/spark-2159c945-ced0-4794-a5b6-7b6a63fb0c2b
[2024-11-09T23:15:11.508+0000] {spark_submit.py:634} INFO - 24/11/09 23:15:11 INFO ShutdownHookManager: Deleting directory /tmp/spark-2159c945-ced0-4794-a5b6-7b6a63fb0c2b/pyspark-8304f097-98e3-4e0f-b9ab-a842e86a7c35
[2024-11-09T23:15:11.513+0000] {spark_submit.py:634} INFO - 24/11/09 23:15:11 INFO ShutdownHookManager: Deleting directory /tmp/spark-980637f6-fe3c-4cbc-a30c-8147d5069890
[2024-11-09T23:15:11.562+0000] {taskinstance.py:3310} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 176, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 560, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark-master:7077 --jars /opt/***/dags/scripts/jars/hadoop-azure-3.4.1.jar,/opt/***/dags/scripts/jars/azure-storage-8.6.6.jar,/opt/***/dags/scripts/jars/jetty-server-9.4.45.v20220203.jar,/opt/***/dags/scripts/jars/jetty-util-9.4.45.v20220203.jar,/opt/***/dags/scripts/jars/jetty-util-ajax-9.4.45.v20220203.jar --name spark_submit_bronze_task --deploy-mode client /opt/***/dags/scripts/bronze_layer.py. Error code is: 1.
[2024-11-09T23:15:11.566+0000] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=openbrewerydb_dag, task_id=spark_submit_bronze_job, run_id=manual__2024-11-09T22:55:57.289669+00:00, execution_date=20241109T225557, start_date=20241109T231335, end_date=20241109T231511
[2024-11-09T23:15:11.586+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-11-09T23:15:11.587+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 482 for task spark_submit_bronze_job (Cannot execute: spark-submit --master spark-master:7077 --jars /opt/***/dags/scripts/jars/hadoop-azure-3.4.1.jar,/opt/***/dags/scripts/jars/azure-storage-8.6.6.jar,/opt/***/dags/scripts/jars/jetty-server-9.4.45.v20220203.jar,/opt/***/dags/scripts/jars/jetty-util-9.4.45.v20220203.jar,/opt/***/dags/scripts/jars/jetty-util-ajax-9.4.45.v20220203.jar --name spark_submit_bronze_task --deploy-mode client /opt/***/dags/scripts/bronze_layer.py. Error code is: 1.; 64)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 115, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3004, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3158, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3182, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 176, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 560, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark-master:7077 --jars /opt/***/dags/scripts/jars/hadoop-azure-3.4.1.jar,/opt/***/dags/scripts/jars/azure-storage-8.6.6.jar,/opt/***/dags/scripts/jars/jetty-server-9.4.45.v20220203.jar,/opt/***/dags/scripts/jars/jetty-util-9.4.45.v20220203.jar,/opt/***/dags/scripts/jars/jetty-util-ajax-9.4.45.v20220203.jar --name spark_submit_bronze_task --deploy-mode client /opt/***/dags/scripts/bronze_layer.py. Error code is: 1.
[2024-11-09T23:15:11.645+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-11-09T23:15:11.658+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
